lib/em/io_streamer.rb

```diff
@@ -0,0 +1,51 @@
+# similar to EventMachine::FileStreamer, but for any IO object
+module EventMachine
+  class IOStreamer
```

Comment:
It's a (nearly) complete rewrite of `EventMachine::FileStreamer` (https://github.com/eventmachine/eventmachine/blob/master/lib/em/streamer.rb#L22), with the same options and semantics, but taking an `IO` object instead of a file path.  Ideally, this would go back into Eventmachine, but I thought it could live here for now.

Reply:
Interesting. We can land this here, but I'd be nice to upstream it.. Do you have a PR open on EM?

Reply:
I do now: https://github.com/eventmachine/eventmachine/pull/855

Reply:
👍